demos/common/cpp/pipelines/include/pipelines/async_pipeline.h

```diff
@@ -83,4 +87,6 @@ class AsyncPipeline {
     std::exception_ptr callbackException = nullptr;
 
     std::unique_ptr<ModelBase> model;
+    PerformanceMetrics inferenceMetrics;
+    PerformanceMetrics preprocessMetrics;
```

Comment:
`PerformanceMetrics` with its current implementation shouldn't be used that way: `PerformanceMetrics` collect latency and throughput, but the latter is discarded. The same is about `ImagesCapture`.

Reply:
Throughput is meaningless here, but the metrics class itself is useful. So reusing it gives more profit than overhead costs brought by throughput calculation inside it. 
I can make throughput calculation configurable in separate patch, but please leave this code here for consistency.

Reply:
> So reusing it gives more profit than overhead costs brought by throughput calculation inside it.

This is not about performance. This is about design.

> I can make throughput calculation configurable in separate patch

1. Not only throughput. This was just what I came up with first. What you report can be collected using just two fields: sum time of all calls and number of that calls. Having complex `PerformanceMetrics` is overkill. BTW why isn't latency reported on image? Having this will at least deny this argument.

2. What's the point in such split? I believe whole metric calculation modifications should be in a separate patch then.

> please leave this code here for consistency

Consistency with what?

Reply:
Last time we had same conversation you were insisting on reusing PerformanceMetrics for measurements. For consistency with other samples code. Now you want to avoid such reuse. PerformanceMetrics allow to get current latency values (calculated over window) along with average values. Using it here is not overkill as long as it is already implemented and it doesn't affect overall performance. I can put these measurements in separate patch, but it would require extra work to do - do we really need it ?

Reply:
> Last time we had same conversation you were insisting on reusing PerformanceMetrics for measurements. For consistency with other samples code.

I claim that you misinterpret the story. I remember myself asking not to touch overall latency, which you were removing. And speaking about extra metrics I didn't insist on using `PerformanceMetrics` in that particular way you have now. But if you provide proofs, I won't turn down on my words. I'm not even sure I was asking to completely remove them at this point. For example when you introduced inference metric back then I just commented that time_point is placed in the middle of callback which doesn't make sense. Then you started complaining about review process and only then it was deiced that you can't touch metrics at all to keep PR simple. Since then I could comment that you can't modify metrics  because I was reminding you the decision. And yet here you are again mixing models architecture and metrics collection.

> I can put these measurements in separate patch, but it would require extra work to do - do we really need it ?

To me this whole update on metrics is extra work from your side. I have perception you are supposed to update model hierarchy here.

> Using it here is not overkill as long as it is already implemented and it doesn't affect overall performance.

You have excluded design, which I believe relates.

Reply:
My solution didn't work out. I can't come up with better design. Excuse me for making you remove the metrics.

Reply:
Sad but true. I'll add them back in separate patch as long as I've already removed them here.